src/components/Department.jsx

```diff
@@ -1,36 +1,74 @@
 import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 import { withRouter } from 'react-router-dom';
+import { apiRequest } from '../api/apiService';
 import { DEPARTMENT_BY_ID_URL, EMPLOYEES_URL } from '../constants/url';
-import { dateFormat } from '../utils/dateFormatter';
+import { AppContext } from '../contexts/AppContext';
+
+import EditDepartmentForm from '../modalForms/EditDepartmentForm';
 import '../styles/Department.css';
 
 class Department extends Component {
+  deleteDeprtment = async () => {
+    const departmentId = this.props.department._id;
+    const [departmentDeletedError, departmentDeleted] =
+      await apiRequest.deleteDepartment(departmentId);
+    if (departmentDeleted) {
+      this.props.remove(departmentId);
+      alert('OK!');
```

Comment:
don't use alerts :)
either implement a user-friendly one or show the error messages somewhere in validation-related places